token/cli/src/output.rs

```diff
@@ -399,6 +481,108 @@ impl fmt::Display for CliTokenAccounts {
+                "    {} {}bps",
+                style("Fee:").bold(),
+                transfer_fee_basis_points
+            )?;
+            writeln_name_value(f, "    Maximum fee:", &maximum_fee.to_string())?;
+            writeln_name_value(
+                f,
+                "    Config authority:",
+                transfer_fee_config_authority
+                    .as_ref()
+                    .unwrap_or(&String::new()),
+            )?;
+            writeln_name_value(
+                f,
+                "    Withdrawal authority:",
+                withdraw_withheld_authority
+                    .as_ref()
+                    .unwrap_or(&String::new()),
+            )?;
+            writeln_name_value(f, "    Withheld fees:", &withheld_amount.to_string())
+        }
+        UiExtension::TransferFeeAmount(UiTransferFeeAmount { withheld_amount }) => {
+            writeln_name_value(f, "  Transfer fees withheld:", &withheld_amount.to_string())
+        }
+        UiExtension::MintCloseAuthority(UiMintCloseAuthority { close_authority }) => {
+            writeln_name_value(
+                f,
+                "  Close authority:",
+                close_authority.as_ref().unwrap_or(&String::new()),
+            )
+        }
+        UiExtension::ConfidentialTransferMint(_) => unimplemented!(),
+        UiExtension::ConfidentialTransferAccount(_) => unimplemented!(),
+        UiExtension::DefaultAccountState(UiDefaultAccountState { account_state }) => {
+            writeln_name_value(f, "  Default state:", &format!("{:?}", account_state))
+        }
+        UiExtension::ImmutableOwner => writeln!(f, "  {}", style("Immutable owner").bold()),
+        UiExtension::MemoTransfer(UiMemoTransfer {
+            require_incoming_transfer_memos,
+        }) => writeln_name_value(
+            f,
+            "  Transfer memo:",
+            if *require_incoming_transfer_memos {
+                "Required"
+            } else {
+                "Not required"
+            },
+        ),
+        UiExtension::NonTransferable => writeln!(f, "  {}", style("Non-transferable").bold()),
+        UiExtension::InterestBearingConfig(UiInterestBearingConfig {
+            rate_authority,
+            current_rate,
+            ..
+        }) => {
+            writeln!(f, "  {}", style("Interest-bearing:").bold())?;
+            writeln!(
+                f,
+                "    {} {}bps",
+                style("Interest rate:").bold(),
+                current_rate
+            )?;
+            writeln_name_value(
+                f,
+                "    Rate authority:",
+                rate_authority.as_ref().unwrap_or(&String::new()),
+            )
+        }
+        // ExtensionType::Uninitialized is a hack to ensure a mint/acocunt is never the same length as a multisig
+        UiExtension::Uninitialized => Ok(()),
+        UiExtension::UnparseableExtension => {
+            panic!("Failed to parse extension; token data may be malformed")
```

Comment:
this seemed more reasonable than returning an error, because this is for a `Display` impl, which expects to see `std::fmt::Error`, but this isnt a format error...

Reply:
If someone has an old version of the token-cli, and we've since added some new extensions, and they try to fetch their account, this could cause a panic, which would stink.  How about we print an error message that an unparseable extension was found, and that they likely need to update `spl-token-cli` to see it?

Reply:
ahh i didnt think of adding more extensions youre right, in my head this case was a just-in-case that was never hit